.github/workflows/aod-iam.yml

```diff
@@ -0,0 +1,51 @@
+# Workflow that validates and handles AOD IAM request.
+name: 'aod-iam'
+
+on:
+  pull_request:
+    types: ['opened']
+    branches:
+      - 'main'
+  pull_request_review:
+    types: ['submitted']
+
+jobs:
+  # Check if it is an AOD request
+  is_iam:
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-checker.yml@main' # Since it does not have any access to github context, directly ping the tag/ref here.
+    with:
+      path: 'iam.yaml'
```

Comment:
Do we need another workflow? I think in the same iam workflow, we can check if `iam.yaml` exist. If not, just skip.

Reply:
The always-fail job (to prevent merging) relies on this checker.

Reply:
Can we merge this job with always-fail?

Reply:
No, because jobs like validate and handle that depend on this will not run because this job will alway fail.